### PR TITLE
HUD overlay fixes and improvements

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
The VR HUD overlay has been fixed and improved:
- Overlay surface is rendered a bit further away (1.5m instead of 1.0m). This makes it easier on the eyes.
- The horizontal size of the overlay surface has been reduced from 270 deg to 180 deg. This means that it doesn't wrap around "behind" you.
- The physical (real world) distance required to move before the HUD overlay is recentered on your avatar has been reduced from 0.99m to 0.33m. This makes the HUD always be reasonably centred on your view rather than possibly being very offset at times.
- Reorienting the overlay surface to better suit your current HMD orientation (i.e., be reasonably well centred on your current view) has been enabled. This means that the HUD overlay can no longer be behind you.
- Fix un-pinned HUD content (e.g., Web browser window) displaying only every second recentering.
- Fix laser intersection with HUD overlay dialogs / windows: laser intersects exactly per dialog position and size. This, for example, means that you can now drag dialogs using their top margin.
- Fix the position of the dialog / window resize outline, displayed when resizing a dialog / window - in both desktop and HMD mode.

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
### Testing

HMD mode ...

The position and orientation of the HUD overlay can most readily be seen if you display the chat bar down the bottom of the screen. (Press Enter to display.)
The browser window (Ctrl-B) can also be useful for testing.

1) Test that the HUD overlay repositions to be centered on your avatar as you move about your physical VR play space.

2) Test that the HUD overlay reorients to be reasonably in front of the direction you're facing. (Rotate your head about the y axis and the overlay will jump to follow your gaze as it reaches the edge of the overlay.)

3) Test laser on HUD items:
- Laser end point coincides with HUD items.
- Laser hovers controls on HUD surface (e.g., "x" button in script debug window).
- Laser click works on HUD surface (e.g., click-and-drag script debug window, "x" button in script debug window closes window).

4) Test mouse pointer on HUD surface:
- ... <per laser testing>

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-65ec822.tar_cbef9d29.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-65ec822.tar_cbef9d29.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-65ec822_cbef9d29.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-65ec822_cbef9d29.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-65ec822_cbef9d29.exe

File: interface/src/ui/OverlayConductor.cpp [hunk 1]
```diff
@@ -96,24 +97,24 @@ void OverlayConductor::update(float dt) {
...
-            shouldRecenter = true;
-        }
-    } else {
-        if (_hmdMode && headOutsideOverlay()) {
-            _suppressedByHead = true;
-        }
+    bool shouldRecenter = false;
+    if (initiateRecenter || _suppressedByHead) {
+        _suppressedByHead = !updateAvatarIsAtRest();
+        shouldRecenter = !_suppressedByHead;
```
Comment (HifiExperiments) [inline interface/src/ui/OverlayConductor.cpp:108]: this logic seems very slightly different than the old code: if _suppressedByHead is false, the old code wouldn't call updateAvatarIsAtRest().  is that ok?

also this code is a little confusing with both initiateRecenter and shouldRecenter, but other than that looks good to me

**Review by HifiExperiments — CHANGES_REQUESTED** [review]

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp [hunk 2]
```diff
@@ -384,9 +384,9 @@ bool CompositorHelper::calculateRayUICollisionPoint(const glm::vec3& position, c
     glm::vec3 localPosition = transformPoint(worldToUi, position);
     glm::vec3 localDirection = glm::normalize(transformVectorFast(worldToUi, direction));
 
-    const float UI_RADIUS = 1.0f;
+    const float UNIT_RADIUS = 1.0f;
     float intersectionDistance;
-    if (raySphereIntersect(localDirection, localPosition, UI_RADIUS, &intersectionDistance)) {
+    if (raySphereIntersect(localDirection, localPosition, UNIT_RADIUS, &intersectionDistance)) {
```
[the remark this answers is not in the document]
Comment (ctrlaltdavid) [reply libraries/display-plugins/src/display-plugins/CompositorHelper.cpp:389]: No, it needs to be UNIT_RADIUS. The HUD_RADIUS is applied to the `uiToWorld` transform applied in the next line:
```
result = transformPoint(uiToWorld, localPosition + localDirection * intersectionDistance);
```

File: interface/src/ui/OverlayConductor.cpp (hunk 1 above)
Comment (ctrlaltdavid) [reply interface/src/ui/OverlayConductor.cpp:108]: The logic had to change as part of fixing the HUD content displaying only every second HUD recentering. Not calling 

`initiateRecenter` is used to signal that the HUD should be recentered because the head is not centered in the HUD.
But updating the HUD position is suppressed if the head is moving, until such time as the head stops moving.

The new code is more straightforward than the old code, I believe.

File: libraries/display-plugins/src/display-plugins/CompositorHelper.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (HifiExperiments) [reply libraries/display-plugins/src/display-plugins/CompositorHelper.cpp:389]: 👍

File: interface/src/ui/OverlayConductor.cpp (hunk 1 above)
Comment (HifiExperiments) [reply interface/src/ui/OverlayConductor.cpp:108]: 👍

**Review by HifiExperiments — APPROVED** [review]

**Comment by SilverfishVR** [comment]
Overall there are improvements and nothing seems to have broken, the hud still has issues and bugs though.
random observations.
* recentering and reorientation is a lot better, I think I would prefer it if recenter worked the same for real world movement as it does for controller movement, being locked to avatar Y orientation/position instead of snapping.
* the scale and distance looks good to me.
* Intersecting (laser) the window border (close, pin, resize) still only works sporadically, it initially worked for me but now it seems mostly broken (as is normal).
* scrollbars in hud windows seems to move the window when lasered, instead of actual scrolling.
* the microphone overlay always work, I can move and click it as expected 😃 
* sometimes, the recenter is slow (or glitchy) so if I turn around, the windows may disappear for a 2-5 seconds 🤔 
* we need to figure out what the pin is supossed to do.

tl:dr the changes in this PR all seems to work 👍  but more fixes (or redesign) is needed to make it good

**Comment by ctrlaltdavid** [comment]
Laser intersecting is worse than before. Not obvious why but I'm working on it.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-fd9702e_b865e9bc.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-fd9702e.tar_b865e9bc.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-fd9702e_b865e9bc.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-fd9702e.tar_b865e9bc.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-fd9702e_b865e9bc.exe

**Comment by SilverfishVR** [comment]
For the first time ever, laser intersecting works as expected now 👍

* if the windows are supposed to disappear when moving if not pinned that still don't work, not sure I miss it though 🤷 
* the scrollbar on the browser works, the one on the debug window does not, it grabs and moves instead.
* the windows still disappear for a few seconds sometimes, they always come back quickly so definitely improved.

All in all, this PR does indeed fix & improve the HUD so I approve 👍 And thank you for the contribution 😄

**Comment by ctrlaltdavid** [comment]
The window disappearing sometimes for a few seconds is the windows disappearing when (Interface thinks) you've moving. Windows that are pinned don't disappear, though.

Scrollbar on the debug window, I'll have a look at.

**Comment by ctrlaltdavid** [comment]
I've created a separate issue for the script log window's scrollbar: https://github.com/kasenvr/project-athena/issues/892

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-8e70e1b.tar_b865e9bc.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-8e70e1b.tar_b865e9bc.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-8e70e1b_b865e9bc.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-8e70e1b_b865e9bc.dmg

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-8e70e1b_b865e9bc.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Quest-Alpha-PR838-8e70e1b_b865e9bc.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR838/Vircadia-Alpha-PR838-8e70e1b_b865e9bc.exe

**Review by HifiExperiments — APPROVED** [review]
